src/uri.cr

```diff
@@ -256,10 +256,10 @@ class URI
     URI::Parser.new(raw_url).run.uri
   end
 
-  # URL-decode a `String`.
+  # URL-decodes *string*.
```

Comment:
Why specifically making this non-contextual? Unlike the other places of the standard library where mostly contextual descriptions are used.
"A string" sounds like it decodes *any* string, no specific one. But that's not true. It URL-decodes the given *string*, no other one.

Reply:
> other places of the standard library where mostly contextual descriptions are used.

I'd like to improve them all, but can't do everything at once.

> Why not being specific and clear?

"URL-decodes *string*." by itself is neither specific nor clear.

Reply:
Well unless one doesn't know that an *italicized* word 99% of the time refers to an argument (which one becomes aware of quite fast after you read some documentation), this is quite clear. It says that the argument `string` you passed to the method will be URL-decoded, and just that, nothing else and only that one string. And with specific and clear I don't mean an intentionally long-written sentence like "This method Uniform Resource Locator-decodes the argument `string`". "URL-decodes *string*" is clear as well.

In every case, "URL-decodes *string*" is more specific than "URL-decodes a `String`".

Reply:
I think the documentation should be written in a way that you can tell someone about it.

If you ask "What does URI.unescape" do and I reply "It url decodes string"... it sounds strange and grammatically incorrect. But "it urls decodes a string" or "it urls decodes the given \*string\*" sounds better.

Reply:
Yeah if you would speak "URL-decodes *string*" aloud then that would sound weird and incorrect. But I understood it the way that in your head you are supposed to replace the *string* by the string you gave. So if you pass `"hello"`, you should understand the documentation as "URL-decodes `"hello"`".
Here's an example where that seems to be the case: https://crystal-lang.org/api/0.28.0/String.html#%2A%28times%3AInt%29-instance-method
> Makes a new `String` by adding *str* to itself *times* times.

"to itself *times* times." doesn't make any sense unless you replace *times* by the actual number you passed. So for example "to itself 5 times."

But I would be fine with `It URL-decodes the given *string*"`.
I really think there should be a clear specification on https://crystal-lang.org/reference/conventions/documenting_code.html about this. Otherwise incosistencies like this happen because nobody really knows the right way.

Reply:
Yes, the description of a method should be understandable when spoken aloud, and work independent of the context.

Reply:
I'm fine with `URL-decodes the given string`, but I'd prefer `URL-decodes a string` because it's more concise. The fact, that the mentioned string is given to the method is obvious. It doesn't work on some random string from a magical source of strings.